app/controllers/miq_ae_customization_controller/dialogs.rb

```diff
@@ -1316,8 +1316,8 @@ def dialog_set_record_vars(dialog)
         end
       end
 
-      if dialog.dialog_fields.blank?
-        raise "Dialog must have at least one Element"
+      if !dialog.dialog_valid?
+        raise dialog.dialog_validate_error_message
```

Comment:
@lfu I want to avoid having to call a separate method to validate the dialog.  This should all work off of the `save!` call below, then these lines would not be needed.